### PR TITLE
Switch from bcrypt to bcryptjs to keep compatibility with Raspberry Pi and a Dockerfile-armhf bonus

### PR DESCRIPTION
**Pull request by fluxtendu**
Because bcrypt refused to install/compile on the docker image on my raspberry pi, I have found this workaround.

I'm not a programmer and I don't know the implications but it should interest some people...

### CODE REVIEW

**Comment by Tzahi12345** [comment]
Nice work!

Have you tested this? You can create a user/set your admin password, and in your `appdata/users.json` you should see the hashed password. Let me know what you see! (you don't need to paste all of it, just some to make sure it looks like a normal hash and isn't `null` or something like that)

**Comment by fluxtendu** [comment]
yes it works, my users.json:
'''
{
  "jwt_secret": "cc999c30... ",
  "users": [
    {
      "name": "admin",
      "uid": "admin",
      "passhash": "$2a$10$...",
      "files": {
        "audio": [],
        "video": []
      },
      "playlists": {
        "audio": [],
        "video": []
      },
      "subscriptions": [],
      "created": 1594767635111,
      "role": "admin",
      "permissions": [],
      "permission_overrides": []
    }
  ],
  "roles": {
    "admin": {
      "permissions": [
        "filemanager",
        "settings",
        "subscriptions",
        "sharing",
        "advanced_download",
        "downloads_manager"
      ]
    },
    "user": {
      "permissions": [
        "filemanager",
        "subscriptions",
        "sharing"
      ]
    }
  }
}
'''

**Comment by Tzahi12345** [comment]
That's enough for me! Here's the main difference between bcrypt and bcrypt.js if anyone's interested: https://github.com/kelektiv/node.bcrypt.js/issues/705#issuecomment-463492202

TLDR is that bcrypt.js is pure js, and can therefore be used anywhere. Bcrypt (no js) relies on compiled C++ to run, which seems to be a problem on non-x86 systems.

I'm on board with this, and the new Dockerfile is a big plus. I should be able to do an autobuild for this too, meaning there can be a `nightly-arm` and `nightly` tags for ARM and x86 respectively (the latter already exists).

If anyone wants to chime in/test, feel free. Otherwise, I'll do some quick testing tonight and merge.

**Comment by fluxtendu** [comment]
I'm glad that you like my pull request, and even more that you consider to make an arm build.

And since I haven't said it yet, thank you for developing this project!

**Comment by Tzahi12345** [comment]
Thank *you* for such an awesome PR! Getting official ARM support will help a ton of people, no doubt. Just did a quick test, old passwords still work and new ones work well too. Merging!
